enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,337 @@
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for heterogeneous
+payloads which CVO will need to look at to make upgrade recommendations. There are a few upgrade scenarios to consider with the addition of
+heterogeneous payloads:
+
+- heterogeneous -> heterogeneous payload upgrade
+- homogeneous -> heterogeneous payload upgrade
+- heterogeneous -> homogeneous payload upgrade
+
+The support for homogeneous payloads will exist so customers running disconnected homogeneous clusters do not incur the cost of mirroring the
+additional architectures if they were to use heterogeneous payloads. This means that in order for CVO to find the current release in the graph,it
+will have to know about the existing homogeneous graph as well in addition to the new heterogeneous graph. There are few options on the best way
+to make update recommendations especially in the case of upgrading from a homogeneous to a heterogeneous payload:
+
+- Have the heterogeneous graph include the homogeneous releases so CVO can make recommendations based on the current homogeneous release image.
+This is complicated by the fact that there will be multiple releases with the same version but different payloads. The trigger to update to a
+heterogeneous payload would be an option provided by the administrator through either the ClusterVersion or Infrastructure API to "move" to the
+heterogeneous graph.
+- Have both the graphs be completely independent of each other and have an option in one of the aforementioned APIs to make the cluster heterogeneous
+and CVO would request the heterogeneous graph, find the current release version and update to it.
+
+These options are under discussion and a finalized proposal will be put forth in the ticket for upgrades referenced above along with the API that would
+change to include the option to indicate a move to the heterogeneous payload.
+
+Cincinnati needs to [support][OCPPLAN-7643] parsing manifestlist images to be able to upgrade to a heterogeneous architecture payload.
+
+### Imagestreams
+
+Imagestreams need to better support manifestlist images. As of today they do recognize manifestlists, but it only imports images matching the control
+plane architecture. The idea here would be to import the manifest listed image in its entirety. This also has other implications such as the fact that
+the internal registry does not support manifest listed images. So, with full manifestlist support in imagestreams, pullthrough is not possible for
+manifestlists. We would ask users to use quay.io rather than use the internal registry if full manifestlist support is needed.
+
+The oc commands like debug, must-gather etc use the imagestreams and therefore would use an x86 image to deploy on the worker nodes which would fail
+with an ARM worker. As a workaround, they can be provided the manifest listed image with the `--image` parameter which would allow the commands to
```

Comment:
```suggestion
with an ARM worker. As a workaround until imagestreams support importing all the architectures, they can be provided the manifest listed image with the `--image` parameter which would allow the commands to
```